includes/class-plugin.php

```diff
@@ -665,38 +666,68 @@ public function render_admin_bar( $wp_admin_bar ) {
             return;
         }
 
+        $html = <<<HTML
+            <style>
+                #wpadminbar ul li.redis-cache-error { background: #c00; }
+                #wpadminbar:not(.mobile) .ab-top-menu > li.redis-cache-error:hover > .ab-item { background: #b30000; color: #fff; }
+            </style>
+            <script>
+                document.querySelector('#wp-admin-bar-redis-cache-flush > a')
+                    .addEventListener('click', async function (event) {
+                        event.preventDefault();
+
+                        var node = document.querySelector('#wp-admin-bar-redis-cache');
+                        var textNode = document.querySelector('#wp-admin-bar-redis-cache > .ab-item:first-child');
+
+                        node.classList.remove('hover');
+                        textNode.innerText = 'Flushing cache...';
+
+                        try {
+                            var response = await fetch(qm_l10n.ajaxurl, {
+                                method: 'POST',
+                                body: 'action=flush_cache&nonce=...'
```

Comment:
We need to embed a `nonce` here, ideally using PHP: https://developer.wordpress.org/reference/functions/wp_create_nonce/